staubli_tx2_60_support/urdf/tx2_60_macro.xacro

```diff
@@ -0,0 +1,239 @@
+          <inertia ixx="0.02469218" ixy="0.00032528" ixz="0.00063194"
+                                    iyy="0.02222651" iyz="-0.00155762"
+                                                     izz="0.01657366"/>
+      </inertial>
+    </link>
+    <link name="${prefix}link_4">
+      <visual>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://staubli_tx2_60_support/meshes/tx2_60/visual/link_4.stl"/>
+        </geometry>
+        <xacro:material_staubli_ral_melon_yellow />
+      </visual>
+      <collision>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://staubli_tx2_60_support/meshes/tx2_60/collision/link_4.stl"/>
+        </geometry>
+      </collision>
+      <inertial>
+          <mass value="3.48055000"/>
+          <origin xyz="-0.00134244 -0.00000186 0.28816464" rpy="0.0 0.0 0.0"/>
+          <inertia ixx="0.03950054" ixy="0.00000004" ixz="-0.00063419"
+                                    iyy="0.03813828" iyz="-0.00000002"
+                                                     izz="0.00610289"/>
+      </inertial>
+    </link>
+    <link name="${prefix}link_5">
+      <visual>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://staubli_tx60_support/meshes/tx60/visual/link_5.stl"/>
+        </geometry>
+        <xacro:material_staubli_ral_grey_aluminium />
+      </visual>
+      <collision>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://staubli_tx60_support/meshes/tx60/collision/link_5.stl"/>
+        </geometry>
+      </collision>
+      <inertial>
+          <mass value="0.16254714"/>
+          <origin xyz="-0.00000105 0.00000000 0.01268974" rpy="0.0 0.0 0.0"/>
+          <inertia ixx="0.00012557" ixy="0.00000000" ixz="0.00000000"
+                                    iyy="0.00013289" iyz="0.00000000"
+                                                     izz="0.00006342"/>
+      </inertial>
+    </link>
+    <link name="${prefix}link_6">
+      <visual>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://staubli_tx60_support/meshes/tx60/visual/link_6.stl"/>
+        </geometry>
+        <xacro:material_staubli_ral_grey_aluminium />
+      </visual>
+      <collision>
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <geometry>
+          <mesh filename="package://staubli_tx60_support/meshes/tx60/collision/link_6.stl"/>
+        </geometry>
+      </collision>
+      <inertial>
+          <mass value="0.00820000"/>
+          <origin xyz="-0.00022629 0.00000000 -0.00597808" rpy="0.0 0.0 0.0"/>
+          <inertia ixx="0.00000096" ixy="0.00000000" ixz="0.00000000"
+                                    iyy="0.00000093" iyz="0.00000000"
+                                                     izz="0.00000154"/>
+      </inertial>
```

Comment:
These values are very small. I've seen reports of Gazebo becoming unstable (or: it's dynamics simulation becoming unstable) with such small values.

Reply:
> A few in-line comments.
> 
> One additional one: the meshes seem rather heavily decimated, which leads to some artefacts (almost). Did you do this on purpose? Especially `link_4` seems like it could use some more faces, which would be OK.

How big can the visual meshed be? Is there a rule of thumb? thank you

Reply:
I would say for `visual` meshes: between 200 to 300/400KB. It depends a bit on which mesh it is.

`link_6` fi will always be much smaller. But `link_4` and `link_3` typically have many doubly curved surfaces, so they'll need a bit more KBytes.

Reply:
Thank you for your fast answer!